python/golem_task_api/apputils/app_definition.py

```diff
@@ -0,0 +1,114 @@
+from distutils import cmd
+import hashlib
+import json
+import shutil
+
+from pathlib import Path
+from typing import Dict, Any
+
+from dataclasses import dataclass, field
+from dataclasses_json import dataclass_json, config
+from marshmallow import fields as mm_fields, validate
+
+AppId = str
+
+
+@dataclass_json
+@dataclass
+class AppDefinitionBase:
+    name: str
+    requestor_env: str
+    requestor_prereq: Dict[str, Any] = field(metadata=config(
+        mm_field=mm_fields.Dict(keys=mm_fields.Str())
+    ))
+    max_benchmark_score: float
+    version: str = '0.0'
+    description: str = ''
+    author: str = ''
+    license: str = ''
+    market_strategy: str = mm_fields.Str(
+        validate=validate.OneOf(["brass", "wasm"]),
+        default='brass'
+    )
+
+    @property
+    def id(self) -> AppId:
+        return hashlib.blake2b(  # pylint: disable=no-member
+            self.to_json().encode('utf-8'),
+            digest_size=16
+        ).hexdigest()
+
+    @classmethod
+    def from_json(cls, json_str: str) -> 'AppDefinitionBase':
+        raise NotImplementedError  # A stub to silence the linters
+
+    def to_json(self) -> str:
+        raise NotImplementedError  # A stub to silence the linters
+
+
+def save_app_to_json_file(app_def: AppDefinitionBase, json_file: Path) -> None:
+    """ Save application definition to the given file in JSON format.
+        Create parent directories if they don't exist. """
+    try:
+        json_file.parent.mkdir(parents=True, exist_ok=True)
+        json_file.write_text(app_def.to_json())
+    except OSError:
+        msg = f"Error writing app definition to file '{json_file}."
+        print(msg)
```

Comment:
Please, don't use `print` in library code. If someone wants to use this function in another way, they might not be happy. Better use a logger – it's configurable.

Reply:
good one! was thinking of using `self._logger` like we did in other places, but this is not a class function.
Maybe printing is not needed at all since it raises an exception.
What do you think? add `self._logger` or remove the print?

Reply:
I'm not sure about that. As far as I understand it's meant to be run in `setup.py` and in that case it's ok to print.

Reply:
this function is mainly used for setup.py, but also to store pre-baked app definitions when golem starts for the first time. 
i would like to remove this extra step and save the app_definition.json in the golem binary, instead of generating it again.

Reply:
Then I would opt for some kind of logger. There is also another print at the end of this file.

Reply:
ok, will do :)

that last print is only used from `setup.py`, but when adding a logger i'll convert them both